R/fit_model.R

```diff
@@ -0,0 +1,42 @@
+#' Fit the RtGam model with {mgcv}
+#'
+#' Use the pre-prepared model dataset and formula. Supply warnings as needed
+#' @noRd
+fit_model <- function(data, formula, backend) {
+  args <- args_constructor(data, formula, backend)
+  call <- call_constructor(backend)
+
+  do.call(
+    call,
+    args
+  )
+}
+
+args_constructor <- function(data, formula, backend) {
+  backend_agnostic_args <- list(
+    formula = formula,
+    data = data,
+    # Negative binomial family with overdispersion param estimated
+    family = "nb"
+  )
+  if (backend == "gam") {
+    backend_specific_args <- list(
+      method = "REML"
+    )
+  } else if (backend == "bam") {
+    backend_specific_args <- list(
+      method = "fREML",
+      discrete = TRUE
+    )
+  } else {
+    cli::cli_abort("Other backends not yet implemented")
```

Comment:
something something s3 something something modularity.

Reply:
Lets move this one to https://github.com/CDCgov/cfa-gam-rt/issues/21